static/js/stream_list.js

```diff
@@ -121,63 +103,40 @@ exports.build_stream_list = function () {
         return;
     }
 
-    streams = filter_streams_by_search(streams);
+    // The main logic to build the list is in stream_sort.js, and
+    // we get thre lists of streams (pinned/normal/dormant).
+    var stream_groups = stream_sort.sort_groups(get_search_term());
+
+    if (stream_groups.same_as_before) {
+        return;
+    }
```

Comment:
I guess probably the `remove_sidebar_row` case is one where we might just want to do a full rerender, so that we can correctly handle things like the bar between inactive streams and active streams...